nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -188,28 +211,207 @@ def apply(
-            modified_model = algorithm.apply(modified_model, modified_model_graph, statistic_points)
-            modified_model_graph = NNCFGraphFactory.create(modified_model)
+            model = algorithm.apply(model, graph, statistic_points)
+            graph = NNCFGraphFactory.create(model)
         # building the model graph is not required after the last algorithm
-        modified_model = self.algorithms[-1].apply(modified_model, modified_model_graph, statistic_points)
+        model = self.algorithms[-1].apply(model, graph, statistic_points)
+
+        return model
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        model_copy = copy_model(model)
+        backend = get_backend(model)
+        if backend in [BackendType.ONNX, BackendType.TORCH]:
+            return self._apply(model_copy, graph, statistic_points, dataset)
+        self._set_backend_entity(backend)
+        if not self._has_if_op(graph, self._backend_entity.if_node_metatype):
+            return self._apply(model_copy, graph, statistic_points, dataset)
+        nncf_logger.info("The model has If operations. The iteratively each body of If operations will be quantized.")
+        quantized_model, _ = self._dfs_quantize_models(model_copy, graph, dataset, statistic_points, 0)
+        return quantized_model
+
+    def _make_dataset_for_child_model(
+        self,
+        engine: Engine,
+        calibration_dataset: Dataset,
+        if_cond_input_name: str,
+        child_model_input_names: List[str],
+        if_submodel_condition: bool,
+        model_cnt: int,
+    ) -> Dataset:
+        """
+        Returns dataset for a child model.
+
+        :param engine: Engine to infer parent model to obtain dataitems for a child dataset.
+        :param calibration_dataset: Dataset to infer parent model.
+        :param if_cond_input_name: Input name of If node condition.
+        :param child_model_input_names: - Names of inputs for child model
+        (should be in the order of passing them to a model).
+        :param if_submodel_condition: If node submodel condition.
+        :param model_cnt: Global counter of a child model.
+        :return Dataset: Dataset for child model.
+        """
+        dataset = []
+        calibration_dataset_size = min(self.subset_size, calibration_dataset.get_length())
+        for input_data in tqdm(
+            islice(calibration_dataset.get_inference_data(), calibration_dataset_size),
+            total=calibration_dataset_size,
+            desc=f"Collect dataset for {model_cnt} model:",
+        ):
+            data_item = []
+            results = engine.infer(input_data)
+            if (if_submodel_condition and results[if_cond_input_name]) or (
+                not if_submodel_condition and not results[if_cond_input_name]
+            ):
```

Comment:
This check is about checking the IF condition bool value on the particular sample. Based on that we collect data or not for particular then/else submodel

Reply:
The question was not about `if_submodel_condition`, but about `if_cond_input_name` since `results` should contain (based on the code) it and not `child_model_input_names` that are really needed. Why so?